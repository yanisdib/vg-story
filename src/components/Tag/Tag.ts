import styled from 'styled-components';


interface TagProps {
    selected: boolean | (() => boolean);
    children: React.PropsWithChildren<string>;
}


export default styled.div`
    width: fit-content;
    height: fit-content;
    padding: 5px 20px;
    background-color: ${({
        selected 
    }: TagProps) => selected ? '#6200ff' : '#e2e0fd'};
    border: 2px solid #6200ff;
    border-radius: 25px;
    color: ${({
        selected 
    }: TagProps) => selected ? '#E0EAFD' : '#6200ff'};;
    font-weight: 500;
    cursor: pointer;
`;