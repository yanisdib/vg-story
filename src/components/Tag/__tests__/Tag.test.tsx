import { render } from '@testing-library/react'

import Tag from '../Tag';


describe('Tag component', () => {
    it('renders correctly without crashing', () => {

        const { container } = render(<Tag selected={false}>Hello</Tag>);

        expect(container).toMatchSnapshot();
    })

    it('has a children text', () => {
        
        const { getByText } = render(<Tag selected={false}>Hello</Tag>);

        expect(getByText('Hello')).toBeInTheDocument;
    })
})