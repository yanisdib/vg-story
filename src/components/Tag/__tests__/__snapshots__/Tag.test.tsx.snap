// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag component renders correctly without crashing 1`] = `
<div>
  <div
    class="sc-bczRLJ kaUByI"
  >
    Hello
  </div>
</div>
`;
